redfish-core/lib/log_services.hpp

```diff
@@ -3411,6 +3436,610 @@ inline void getSystemHardwareIsolationLogService(
+                                     << dbusObjPath.str;
+                    return;
+                }
+
+                redfishUri.replace(
+                    uriIdPos, uriIdPattern.length(),
+                    getIsolatedHwItemId(
+                        sdbusplus::message::object_path(uriIdObj)));
+
+                if (redfishUri.find(uriIdPattern) == std::string::npos)
+                {
+                    nlohmann::json& entryJson =
+                        (entryJsonIdx > 0
+                             ? asyncResp->res
+                                   .jsonValue["Members"][entryJsonIdx - 1]
+                             : asyncResp->res.jsonValue);
+
+                    entryJson["Links"]["OriginOfCondition"] = {
+                        {"@odata.id", redfishUri}};
+                }
+                else
+                {
+                    fillRedfishUriWithIds(asyncResp, entryJsonIdx, dbusObjPath,
+                                          redfishUri);
+                }
+            },
+            objMapperSerName, objMapperObjPath, objMapperIface,
+            "GetSubTreePaths", "/xyz/openbmc_project/inventory", 0,
+            std::get<0>(*parentRedfishUriIt));
+    }
+}
+
+/**
+ * @brief API used to fill the respective redfish uri (which will get by using
+ *        the given dbus object path) into "OriginOfCondition" property of
+ *        LogEntry schema.
+ *
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details in the appropriate entry json object.
+ * @param[in] dbusObjPath - The DBus object path which represents redfishUri.
+ * @param[in] redfishUri - The redfish uri to fill with its "Id".
+ *
+ * @return The redfish response with "OriginOfCondition" property of
+ *         LogEntry schema if success else nothing in redfish response.
+ *
+ * @note - This function will make sure to assign the redfish uri with Ids
+ *         if it contains "<str>".
+ *       - This function will run recursively until finding the redfish uri
+ *         from the list by using the given dbus object path.
+ */
+inline void
+    getRedfishUriFromList(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                          const size_t entryJsonIdx,
+                          const sdbusplus::message::object_path& dbusObjPath,
+                          RedfishUriListType::const_iterator redfishUriIt)
+{
+    if (redfishUriIt == redfishUriList.end())
+    {
+        BMCWEB_LOG_ERROR
+            << "Failed to get Redfish URI for the given DBus object path: "
+            << dbusObjPath.str;
+        return;
+    }
+
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, entryJsonIdx, dbusObjPath,
+         redfishUriIt](const boost::system::error_code ec,
+                       const GetObjectType& objType) mutable {
```

Comment:
what is mutable here?

Reply:
We don't have recursive functions in bmcweb. 
Can we just do this as a for loop?